File: gis-mock-api/src/main/java/com/example/realestate/dto/RealEstateMapDto.java
package com.example.realestate.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * A minimal DTO containing only the fields needed by the map,
 * to allow returning many (e.g. 100k) records efficiently.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RealEstateMapDto {

    private String id;          // Internal DB ID (still used for linking)
    private String mlsNumber;
    private String taxId;

    private String address;
    private String city;
    private String state;
    private String zip;

    private String soldTerms;
    private String soldPrice;

    private Double latitude;
    private Double longitude;

    private String status;
}

File: gis-mock-frontend/src/interfaces/RealEstateMapDto.ts
/**
 * src/interfaces/RealEstateMapDto.ts
 *
 * Updated to include extra fields that we need to edit in the marker pop-up:
 *  - soldTerms, soldPrice, taxId, address, zip
 *  - the rest of the fields remain the same
 */

export interface RealEstateMapDto {
    /** Internal DB ID (still used for linking). */
    id: string;

    /** MLS# as provided by the database. */
    mlsNumber?: string;

    /** Additional fields we need to edit right in the pop-up. */
    soldTerms?: string;
    soldPrice?: string;
    taxId?: string;
    address?: string;
    city?: string;
    state?: string;
    zip?: string;
    status?: string;

    /** Coordinates for placing a marker. */
    latitude?: number | null;
    longitude?: number | null;
}
